File: src/main/java/kr/co/lunatalk/domain/product/dto/ProductWithImagesResult.java
package kr.co.lunatalk.domain.product.dto;

import kr.co.lunatalk.domain.image.domain.Image;
import kr.co.lunatalk.domain.product.domain.Product;

import java.util.List;
import java.util.Map;

public record ProductWithImagesResult(
	List<Product> products,
	Map<Long, List<Image>> imageMap
) {
}

File: src/main/kotlin/kr/co/lunatalk/domain/product/dto/ProductWithImagesResult.kt
package kr.co.lunatalk.domain.product.dto

import kr.co.lunatalk.domain.image.domain.Image
import kr.co.lunatalk.domain.product.domain.Product

data class ProductWithImagesResult(
    val products: List<Product>,
    val imageMap: Map<Long, List<Image>>
)
